enumerables.rb

```diff
@@ -0,0 +1,99 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    length.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    length.times do |i|
+      yield to_a[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all
+    return to_enum(:my_all) unless block_given?
+
+    result = true
+    my_each { |i| result = false unless yield(i) }
+    result
+  end
+
+  def my_any
+    return to_enum(:my_any) unless block_given?
+
+    result = false
+    my_each { |i| result = true if yield(i) }
+    result
+  end
+
+  def my_none
+    return to_enum(:my_none) unless block_given?
+
+    result = true
+    my_each { |i| result = false if yield(i) }
+    result
+  end
+
+  def my_count
+    return to_enum(:my_count) unless block_given?
+
+    result = 0
+    my_each { |i| result += 1 if yield(i) }
+    result
+  end
```

Comment:
- [ ] `my_count` is not working as the original `count` method, also is not passing all test cases. 

### `count` is an Enumerable that has these characteristics:
- Returns the count of elements inside an array/range/hash.
- When a block is given it returns the count of elements that match the condition
```
[1, 2, 3].count { |num| num > 2 }
# 1
```
[Enumerable#count](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-count)